lib/vsc/utils/fancylogger.py

```diff
@@ -188,12 +192,20 @@ def makeRecord(self, name, level, pathname, lineno, msg, args, excinfo, func=Non
             new_msg = msg.encode('utf8', 'replace')
         return logrecordcls(name, level, pathname, lineno, new_msg, args, excinfo)
 
+    def fail(self, message, *args):
+        """Log error message and raise exception."""
+        formatted_message = message % args
+        self.DEFAULT_RAISE_EXCEPTION_LOG_METHOD(formatted_message)
+        raise self.DEFAULT_RAISE_EXCEPTION_CLASS(formatted_message)
+
```

Comment:
Can't you call raiseException inside this function instead of implementing the same thing again?